===================================================================
RCS file: /cvsroot/mozilla/layout/generic nsBlockFrame.cpp,v
retrieving revision 3.856
retrieving revision 3.857
diff -u -r3.856 -r3.857
--- mozilla/layout/generic/nsBlockFrame.cpp	2007/08/14 20:52:21	3.856
+++ mozilla/layout/generic/nsBlockFrame.cpp	2007/08/18 00:06:46	3.857
@@ -4983,8 +4983,17 @@
     return mAbsoluteContainer.RemoveFrame(this, aListName, aOldFrame);
   }
   else if (nsGkAtoms::floatList == aListName) {
-    RemoveFloat(aOldFrame);
-    MarkSameSpaceManagerLinesDirty(this);
+    nsIFrame* curFrame = aOldFrame;
+    // Make sure to delete all the continuations for the float frame
+    // we are removing; this way is a bit messy, but so is the rest of the code.
+    // See bug 390762.
+    do {
+      nsIFrame* continuation = curFrame->GetNextContinuation();
+      nsBlockFrame* curParent = static_cast<nsBlockFrame*>(curFrame->GetParent());
+      curParent->RemoveFloat(curFrame);
+      MarkSameSpaceManagerLinesDirty(curParent);
+      curFrame = continuation;
+    } while (curFrame);
   }
 #ifdef IBMBIDI
   else if (nsGkAtoms::nextBidi == aListName) {

